File: A5/src/stmt/CompStmt.java
package stmt;

import exp.PrgState;
import stack.*;

public class CompStmt implements IStmt
{
	IStmt first;
	IStmt snd;
	
	public CompStmt(IStmt f, IStmt s)
	{
		this.first = f;
		this.snd = s;
	}
	@Override
	public PrgState execute(PrgState state)
	{
		MyIStack<IStmt> stk = state.getExeStack();
		stk.push(snd);
		stk.push(first);
		return state;
	}
	
	public String toString()
	{
		return "(" + first.toString() + "; " + snd.toString() + ")";
	}
}

File: A7/Stmt/CompStmt.cs
using A7.Model;
using A7.Stack;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace A7.Stmt
{
    class CompStmt: IStmt
    {
        IStmt first;
        IStmt snd;

        public CompStmt(IStmt f, IStmt s)
        {
            this.first = f;
            this.snd = s;
        }
        public PrgState Execute(PrgState state)
        {
            MyIStack<IStmt> stk = state.GetExeStack();
            stk.Push(snd);
            stk.Push(first);
            return state;
        }
        public override String ToString()
        {
            return "(" + first.ToString() + "; " + snd.ToString() + ")";
        }
    }
}
